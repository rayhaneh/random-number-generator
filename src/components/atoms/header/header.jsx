import React from 'react';

export const Header = (props) => {
  return (
    <h1 className={'blue'}>
      {props.text}
    </h1>
  );
}
