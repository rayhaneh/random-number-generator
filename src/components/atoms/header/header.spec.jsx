import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import { Header } from './header';

let props;
let comp;

describe('Header Component', () => {
  beforeEach(() => {
    props = {
      text: 'Main Title',
    }
  });

  it('renders properly', () => {
    comp = shallow(<Header {...props} />);
    expect(comp).toMatchSnapshot();
  });
});
