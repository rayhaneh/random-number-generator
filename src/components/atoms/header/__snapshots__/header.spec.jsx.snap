// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    text="Main Title"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Main Title",
      "className": "blue",
    },
    "ref": null,
    "rendered": "Main Title",
    "type": "h1",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Main Title",
        "className": "blue",
      },
      "ref": null,
      "rendered": "Main Title",
      "type": "h1",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
